subsys/net/lib/lwm2m/lwm2m_observation.c

```diff
@@ -0,0 +1,1438 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2018-2019 Foundries.io
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Uses some original concepts by:
+ *         Joakim Eriksson <joakime@sics.se>
+ *         Niclas Finne <nfi@sics.se>
+ *         Joel Hoglund <joel@sics.se>
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_observation
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <fcntl.h>
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/http_parser_url.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/net/lwm2m.h>
+#if defined(CONFIG_LWM2M_DTLS_SUPPORT)
+#include <zephyr/net/tls_credentials.h>
+#endif
+#if defined(CONFIG_DNS_RESOLVER)
+#include <zephyr/net/dns_resolve.h>
+#endif
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_rw_link_format.h"
```

Comment:
Seems like all the seven content format headers can be removed from here too - similarly to the PR #46876